src/features/defi/components/Deposit/PairDepositWithAllocation.tsx

```diff
@@ -0,0 +1,320 @@
+import { Button, Stack, useColorModeValue } from '@chakra-ui/react'
+import type { Asset } from '@shapeshiftoss/asset-service'
+import type { AccountId } from '@shapeshiftoss/caip/dist/accountId/accountId'
+import type { MarketData } from '@shapeshiftoss/types'
+import get from 'lodash/get'
+import { calculateYearlyYield } from 'plugins/cosmos/components/modals/Staking/StakingCommon'
+import type { ControllerProps } from 'react-hook-form'
+import { useController, useForm, useWatch } from 'react-hook-form'
+import { useTranslate } from 'react-polyglot'
+import type { AccountDropdownProps } from 'components/AccountDropdown/AccountDropdown'
+import { Amount } from 'components/Amount/Amount'
+import { AssetIcon } from 'components/AssetIcon'
+import { Allocation } from 'components/DeFi/components/Allocation'
+import { AssetInput } from 'components/DeFi/components/AssetInput'
+import { FormField } from 'components/DeFi/components/FormField'
+import { Row } from 'components/Row/Row'
+import { Text } from 'components/Text'
+import { bnOrZero } from 'lib/bignumber/bignumber'
+
+import { PairIcons } from '../PairIcons/PairIcons'
+
+type DepositProps = {
+  accountId?: AccountId | undefined
+  asset1: Asset
+  asset2: Asset
+  destAsset: Asset
+  // Estimated apy (Deposit Only)
+  apy: string
+  // Use form values to calculate allocation fraction
+  calculateAllocations?(
+    asset: Asset,
+    amount: string,
+  ): { allocationFraction: string; shareOutAmount: string }
+  // Users available amount
+  cryptoAmountAvailable1: string
+  cryptoAmountAvailable2: string
+  // Validation rules for the crypto input
+  cryptoInputValidation1?: ControllerProps['rules']
+  cryptoInputValidation2?: ControllerProps['rules']
+  // Users available amount
+  fiatAmountAvailable1: string
+  fiatAmountAvailable2: string
+  // Validation rules for the fiat input
+  fiatInputValidation1?: ControllerProps['rules']
+  fiatInputValidation2?: ControllerProps['rules']
+  // enables slippage UI (defaults to true)
+  enableSlippage?: boolean
+  // Asset1 market data
+  marketData1: MarketData
+  // Asset2 market data
+  marketData2: MarketData
```

Comment:
Don't pass this down, use `selectMarketDataById` selectors - you have both AssetIDs available here